apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -317,6 +323,59 @@
   };
 
   /**
+   * Returns: { promise, resolve }
+   */
+  LockScreenStateManager.prototype.deferredRequest =
```

Comment:
This can be a classical `Deferred()` and it's type can be checked via `instanceof Deferred`.

Please use the classical Deferred constructor to make the code more readable.

```javascript
function Deferred() {
  this.promise = new Promise((res, rej) => {
    this.resolve = res;
    this.reject = rej;
  });

  return this;
}